src/test/sigencoding_tests.cpp

```diff
@@ -47,13 +53,18 @@ static void CheckSignatureEncodingWithSigHashType(const valtype &vchSig,
     }
 
     for (const SigHashType &sigHash : sigHashes) {
-        // Check the signature with the proper forkid flag.
+        // isTaproot requires Schnorr sigs and BIP341 sighash
+        const bool expectedValid =
+            isTaproot ? is64 && sigHash.hasBIP341() : true;
```

Comment:
This is basically requiring that if it's a BIP341 sighash then it must be a schnorr signature? Is this true in OP_CHECKSIG?

Reply:
No, `isTaproot` is only true for directly spending a taproot using a single signature. Possibly it should be renamed to `isTRKeySpend`